File: app-miyou/src/main/java/com/miyou/app/domain/credit/model/MissionReward.java
package com.miyou.app.domain.credit.model;

import com.miyou.app.domain.mission.model.MissionId;

public record MissionReward(
	MissionId missionId,
	String missionType
) implements CreditSource {

	@Override
	public CreditSourceType sourceType() {
		return CreditSourceType.MISSION_REWARD;
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/credit/model/MissionReward.kt
package com.miyou.app.domain.credit.model

data class MissionReward(
    val missionId: String,
    val missionType: String,
) : CreditSource {
    override fun sourceType(): CreditSourceType = CreditSourceType.MISSION_REWARD

    fun missionId(): String = missionId

    fun missionType(): String = missionType
}
